### DISCUSSION TITLE
How to analyse both heating and cooling supply for the same district area?

### DISCUSSION THREAD

**Discussion by TimMan2020** [architecture-building-systems/CityEnergyAnalyst#3256] [Q&A]
Hi! I’m kind of new to the technicalities of network configurations for district heating and cooling, which lead to some issues with interpreting what exactly is happening in CEA’s thermal network simulation and optimization.

What I see is that in CEA there are two network layout and simulation routines, one for **district heating (DH)** and one for **district cooling (DC)**. Subsequently, the optimization engine can be run separately for either heating or cooling supply. The outputs for DH or DC seem to be completely overwritten in the output data if the two routines are run one after the other. This raised some questions for me:

1. Are the DH and DC optimization routines mutually exclusive, i.e. can I only optimize heating _or_ cooling systems for a given district area? I’ve seen in [Fonseca et al.’s paper](https://www.sciencedirect.com/science/article/pii/S0378778815304199) that both heating and cooling are analysed with CEA. So is there a workaround to analysing both systems, e.g. by copying and saving the outputs for heating, then running cooling, and then adding the numbers for costs and GHG emissions?
2. If the routines are indeed completely detached from one another, am I right in assuming that the network configuration CEA essentially represents is a four-pipe system with two parallel pipes (supply & return) for space/water heating and, separately, two pipes for cold delivery? Likewise, this would imply that centralized heat pumps (HP) in CEA cannot provide simultaneously both heating (from the HP condensers) and cooling (from the HP evaporators) and that storage facilities are separate for heating and cooling, right?
3. What is the _allow-looped-networks_ setting in thermal network layout routine about, does this have any implications for possibly co-optimising heating & cooling supply?

Thanks for your help, much appreciate your work on CEA! 👍 
Tim

**Comment by ShiZhongming** [comment]
@MatNif 
Do you have an immediate answer for this? : )

**Comment by MatNif** [comment]
Hi @TimMan2020 

First of all, I'm terribly sorry to have let you wait for an answer for so long.

As to your questions above:

1. I believe one of the assumptions behind network and optimisation scripts at the time of their creation, was that one would only want to install either a DC or a DH system (with specific use cases in Switzerland and Singapore in mind). The network-script would therefore only take the building's heating (or cooling) demand into account when calculating network layout, pumping requirements, network losses etc. of a DH (or DC) system. In that sense, the DC and DH routines are mutually exclusive indeed. What you described would be a valid workaround if you wanted to get optimal networks for both systems.
2. Mostly correct. Two entirely separate networks are created for DC and DH, so each of the components and pipes would be 'doubly' installed even if there is an overlap. The only small misinterpretation in your assumptions is that the DH system does not take both space & water heating into account. It only takes space heating into account (I believe the original reason for this was that the required temperature levels differ).
3. In short, no; Looped networks don't have implications for a co-optimisation. Looped networks are essentially created in the end, i.e. once the Steiner-tree optimisation for determining the optimal network graph is complete. If I'm not mistaken, this option checks for 'loose' networks nodes (i.e. that are not building or plant nodes) and tries to see if the potential network graph (i.e. streets by default) would allow for some of them to be connected among each other, thus creating loops in parts of the network graph. I believe the idea for this was to see if pumping and thermal losses could be lowered.

**Reply by TimMan2020** [reply]
Dear @MatNif

All clear! Thanks a lot for your comprehensive reply!
